### PR TITLE
Set up a "Hello world" project with two apps

### PR DESCRIPTION
**Pull request by Tekhlay**
### Milestone 💪 

- [x] Create a new React app called 'hello-react-front-end'
- [x] Create the App component with react-router.
- [x] Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- [x] Create a store, an action and a reducer that will connect to you API endpont to get the random greeting.
- [x] Display the random greeting in your Greeting component.
[PR link for Back end](https://github.com/Tekhlay/hello-rails-back-end/pull/1)

### CODE REVIEW

**Comment by Kamba56** [comment]
Hi @Tekhlay

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Kindly add link to the backend repo in the frontend readme.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/redux/messageReducer.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import axios from 'axios';
+
+const initialState = {
+  message_data: '',
+  status: '',
+  error: null,
+};
+
+const FETCH_URL = 'http://localhost:3000/api/v1/messages';
+
```
Comment (Qoosim) [inline src/redux/messageReducer.js]: - [ ] Kindly correct your fetch_url to the one that generates the random greetings in your backend. The correct url should `'http://localhost:3000/api/v1/messages/index'` based on what I saw in your route and what I used to test your api in the rails-backend app.

your-backend

File: src/components/Greetings.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  useEffect(() => {
+    dispatch(getRandomGreeting());
+  }, [dispatch]);
+
+  return (
+    <div className="container">
+      <h3>{messages}</h3>
+    </div>
+  );
```
Comment (Qoosim) [inline src/components/Greetings.js]: - [ ] While trying to run your react app to display the random greetings from your backend, I got blank page. You might need to look into the `cors` in your backend for this to be able to display on another PC

app-behave-1

app-behave-2

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @Tekhlay, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly check the link to the backend from your frontend PR, it should redirect to the backend repo not the backend PR
- [ ] It is also expected to be added to the frontend README file, not to the frontend PR.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**